wordpress-mobile/WordPress-FluxC-Android
# [Detekt Baseline Warnings] Plugins Woo Commerce Module - Resolve/Suppress Complexity & Naming Warnings

Parent #2479
Partially Closes: #2486

This PR resolves/suppresses all `complexity` and `naming` related warnings for the `plugins:woocommerce` module:

- `37` x complexity
    1. `1` x ComplexCondition (Suppress: d4af78b4971a41870dfb488d84957032d08d150d)
    2. `10` x ComplexMethod (Resolve: 28c1c3d45d15d21a882d1df413ddde91ae5b093b + fbe20a26eb7eba54e911a1ec38279f4e530bd976 + fb080616cf0cfb80c1b6e02d84afc03c018f75b1 + Suppress: + ba6729396f3d27a58021b72e92d40cf8474c10f3)
    3. `4` x LargeClass (Suppress: 874a49e3a54b243c91e41c952d56226c56923297)
    4. `10` x LongMethod (Resolve: 721debe369111a66d92a802e2b65f71db53bc89e + 38fdf3b0094c9ec2de2363a7f66d212073f3dc69 + Suppress: 732726889efb9d87311e5ba54503f9676f4c362a)
    5. `6` x LongParameterList (Suppress: 86eb8cdebabfa974fe9e110f7fe88949e334a743)
    6. `6` x NestedBlockDepth (Resolve: ad2e0efead32383109862207c11634f6325aacad + Suppress: cfb3497395b0d436daebc96e4065d96aff549b3e)
- `99` x naming
    1. `9` x ConstructorParameterNaming (Suppress: b02d7ed90baff83bb7a270da786296297a471897)
    2. `1` x MemberNameEqualsClassName (Suppress: 5196de5f84ca59bb4ef95bbbb03687a685506426)
    3. `89` x VariableNaming (Resolve: 6df77dbdff4b3825f444ba3bbf245306253b99c4 + Suppress: 3865d165987092986d60b98e974d9ca755b09af9)

PS.1: I am randomly adding an engineer from the WCAndroid team as the main reviewers, that is, in addition to the @wordpress-mobile/owl-team team itself, since I just want someone from that team to sign-off on that change as well (or additionally to the 🦉 team).

PS.2: I recommend reviewing this PR commit-by-commit.

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough (especially the `detekt` check).
- However, if you really want to be thorough, you could smoke test the `example` and/or the `WCAndroid` apps to verify that everything works as expected.

PS: Please ignore the failing `connected-tests` check as they are unrelated to this PR. This check has been failing consistency for some time now due to tests being flaky or needing to be updated/fixed.

Merge instructions:

- [x] Wait for #2494 to be approved and merged to `trunk`.
